doc/api/http.md

```diff
@@ -2886,15 +2886,15 @@ Accept: text/plain
 To parse the URL into its parts:
 
 ```js
-new URL(request.url, `http://${request.headers.host}`);
+new URL(`http://${req.headers.host}${req.url}`);
```

Comment:
This is not safer than the original, see https://github.com/nodejs/node/issues/52494#issuecomment-2057670240

Reply:
@lpinca is correct. Relying on user headers for any part of the url parsing can lead to the malicious injection of a URL.

I believe the smartest idea would be to just include a note in the documentation that this setup is only an example and shouldn't be used in production for security reasons

Reply:
Sorry, I reviewed without thoroughly analysis. I second @RedYetiDev idea.

Reply:
Nice to see a PR fixing the docs. Maybe the best idea to fix the problem with the `Host` is simply not use it? Maybe something like this is better?

```
new URL(`http://localhost${request.url}`)
new URL(`http://${process.env.HOST ?? 'localhost'}${request.url}`)
```

> I believe the smartest idea would be to just include a note in the documentation that this setup is only an example and shouldn't be used in production for security reasons

I actually read the nodeJS docs as a best practice 🙈 If it is not documented here who should come up with the best possible approach?

Reply:
If we were to go with your approach, we wouldn't need a note because it wouldn't present a security issue anymore.

If the team agrees with your suggestion, you should open a PR for it. With all the work you put into finding and fixing this issue, you should be the one to request it!

Reply:
> If the team agrees with your suggestion, you should open a PR for it.

Sure can do. Waiting for a agree- or disagreement 🙈 

> With all the work you put into finding and fixing this issue, you should be the one to request it!

Team afford. Kudos to @astlouisf and @samhh

Reply:
@RedYetiDev I agree, I can close this one and @mlegenhausen can open a seperate PR to fix this issue